core/controllers/base_test.py

```diff
@@ -312,19 +311,12 @@ def test_renders_error_page_with_iframed(self):
         )
 
     def test_dev_mode_cannot_be_true_on_production(self):
-        # We need to delete the existing module else the re-importing
-        # would just call the existing module.
-        del sys.modules['feconf']
         server_software_swap = self.swap(
             os, 'environ', {'SERVER_SOFTWARE': 'Production'})
         assert_raises_regexp_context_manager = self.assertRaisesRegexp(
             Exception, 'DEV_MODE can\'t be true on production.')
         with assert_raises_regexp_context_manager, server_software_swap:
-            # This pragma is needed since we are re-importing under
-            # invalid conditions. The pylint error messages
-            # 'reimported', 'unused-variable', 'redefined-outer-name' and
-            # 'unused-import' would appear if this line was not disabled.
-            import feconf  # pylint: disable-all
+            importlib.reload(feconf)  # pylint: disable-all
```

Comment:
I think this may need a comment to explain what's going on.

Reply:
@brianrodri ?

Reply:
Added an explanatory comment.